File: Catalog/Examples/Editor/OpenPhoto/FromAppBundle/OpenPhotoFromAppBundle.m
#import "OpenPhotoFromAppBundle.h"
@import PhotoEditorSDK;

@interface OpenPhotoFromAppBundleObjC () <PESDKPhotoEditViewControllerDelegate>

@end

@implementation OpenPhotoFromAppBundleObjC

- (void)invokeExample {
  // Get the URL for a file on the local file system. This could be any file within the app bundle
  // or also a file within the documents or temporary folders for example. This example will use
  // a file from the app bundle.
  // highlight-bundle-url
  NSURL *url = [NSBundle.mainBundle URLForResource:@"LA" withExtension:@"jpg"];
  // highlight-bundle-url

  // Create a `PESDKPhoto` from the video URL.
  // highlight-instantiation
  PESDKPhoto *photo = [[PESDKPhoto alloc] initWithURL:url];

  // Create a default `PESDKConfiguration`.
  PESDKConfiguration *configuration = [[PESDKConfiguration alloc] init];

  // Create a photo editor and pass it the photo and configuration. Make this class the delegate of it to handle export and cancelation.
  PESDKPhotoEditViewController *photoEditViewController = [[PESDKPhotoEditViewController alloc] initWithPhotoAsset:photo configuration:configuration];
  photoEditViewController.delegate = self;
  // highlight-instantiation

  // Present the photo editor.
  photoEditViewController.modalPresentationStyle = UIModalPresentationFullScreen;
  [self.presentingViewController presentViewController:photoEditViewController animated:YES completion:nil];
}

#pragma mark - PESDKPhotoEditViewControllerDelegate

- (BOOL)photoEditViewControllerShouldStart:(PESDKPhotoEditViewController * _Nonnull)photoEditViewController task:(PESDKPhotoEditorTask * _Nonnull)task {
  // Implementing this method is optional. You can perform additional validation and interrupt the process by returning `NO`.
  return YES;
}

- (void)photoEditViewControllerDidFinish:(PESDKPhotoEditViewController * _Nonnull)photoEditViewController result:(PESDKPhotoEditorResult * _Nonnull)result {
  // The image has been exported successfully and is passed as an `NSData` object in the `result.output.data`.
  // To create an `UIImage` from the output, use `[UIImage initWithData:]`.
  // See other examples about how to save the resulting image.
  [self.presentingViewController dismissViewControllerAnimated:YES completion:nil];
}

- (void)photoEditViewControllerDidFail:(PESDKPhotoEditViewController * _Nonnull)photoEditViewController error:(PESDKPhotoEditorError * _Nonnull)error {
  // There was an error generating the photo.
  NSLog(@"%@", error.localizedDescription);
  // Dismissing the editor.
  [self.presentingViewController dismissViewControllerAnimated:YES completion:nil];
}

- (void)photoEditViewControllerDidCancel:(PESDKPhotoEditViewController * _Nonnull)photoEditViewController {
  // The user tapped on the cancel button within the editor. Dismissing the editor.
  [self.presentingViewController dismissViewControllerAnimated:YES completion:nil];
}

@end

File: Catalog/Examples/Editor/OpenPhoto/FromAppBundle/OpenPhotoFromAppBundle.swift
import PhotoEditorSDK
import UIKit

class OpenPhotoFromAppBundleSwift: Example, PhotoEditViewControllerDelegate {
  override func invokeExample() {
    // Get the URL for a file on the local file system. This could be any file within the app bundle
    // or also a file within the documents or temporary folders for example. This example will use
    // a file from the app bundle.
    // highlight-bundle-url
    guard let url = Bundle.main.url(forResource: "LA", withExtension: "jpg") else {
      fatalError("Unable to create URL for specified file.")
    }
    // highlight-bundle-url

    // Create a `Photo` from the photo URL.
    // highlight-instantiation
    let photo = Photo(url: url)

    // Create a photo editor and pass it the photo. Make this class the delegate of it to handle export and cancelation.
    let photoEditViewController = PhotoEditViewController(photoAsset: photo)
    photoEditViewController.delegate = self
    // highlight-instantiation

    // Present the photo editor.
    photoEditViewController.modalPresentationStyle = .fullScreen
    presentingViewController?.present(photoEditViewController, animated: true, completion: nil)
  }

  // MARK: - PhotoEditViewControllerDelegate

  func photoEditViewControllerShouldStart(_ photoEditViewController: PhotoEditViewController, task: PhotoEditorTask) -> Bool {
    // Implementing this method is optional. You can perform additional validation and interrupt the process by returning `false`.
    true
  }

  func photoEditViewControllerDidFinish(_ photoEditViewController: PhotoEditViewController, result: PhotoEditorResult) {
    // The image has been exported successfully and is passed as an `Data` object in the `result.output.data`.
    // To create an `UIImage` from the output, use `UIImage(data:)`.
    // See other examples about how to save the resulting image.
    presentingViewController?.dismiss(animated: true, completion: nil)
  }

  func photoEditViewControllerDidFail(_ photoEditViewController: PhotoEditViewController, error: PhotoEditorError) {
    // There was an error generating the photo.
    print(error.localizedDescription)
    // Dismissing the editor.
    presentingViewController?.dismiss(animated: true, completion: nil)
  }

  func photoEditViewControllerDidCancel(_ photoEditViewController: PhotoEditViewController) {
    // The user tapped on the cancel button within the editor. Dismissing the editor.
    presentingViewController?.dismiss(animated: true, completion: nil)
  }
}
